Program.cs

```diff
@@ -1,2 +1,189 @@
+            }
+            else Console.WriteLine("Your result: {0:0.##}\n", result);
+        }
+
+        private static double CalculateResult(Calculator calculator, double cleanNum1, double cleanNum2, string op)
+        {
+            double result = 0;
+            try
+            {
+                result = calculator.DoOperation(cleanNum1, cleanNum2, op);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Oh no! An exception occurred trying to do the math.\n - Details: " + e.Message);
+            }
+            return result;
+        }
+
+        private static void DisplayMemory(Calculator calculator)
+        {
+            Console.WriteLine("---------Memory--------\n");
+            string memory = string.Empty;
+            int i = 0;
+            foreach (Calculation calc in calculator.CalculationMemory)
+            {
+                memory += $"{i}: {calc}";
+                i++;
+            }
+            Console.WriteLine(memory);
+            Console.WriteLine("To reference memory in your input type 'M' followed by the memory reference (e.g. M2)");
+        }
+
+        private static void DisplayOperatorOptions()
+        {
+            Console.WriteLine("Choose an operator from the following list:");
+            Console.WriteLine("\ta - Add");
+            Console.WriteLine("\ts - Subtract");
+            Console.WriteLine("\tm - Multiply");
+            Console.WriteLine("\td - Divide");
+            Console.WriteLine("\tp - Power");
+            Console.WriteLine("\tsin - Sin(num1)");
+            Console.WriteLine("\tcos - Cos(num1)");
+            Console.WriteLine("\ttan - Tan(num1)");
+            Console.Write("Your option? ");
+        }
+
+        private static void DisplayCleanupOptions()
+        {
+            Console.WriteLine("-----------------------\n");
+            Console.WriteLine("What would you like to do next?:");
+            Console.WriteLine("\tn - Exit");
+            Console.WriteLine("\tc - Clear History");
+            Console.WriteLine("\tOther - Continue calculating!");
+            Console.Write("Your option? ");
+        }
+
+        private static string ValidateOperatorInput()
+        {
+            string op = Console.ReadLine();
+            while (!IsValidOperator(op))
+            {
+                Console.Write("This is not valid input. Please enter one of the above operators: ");
+                op = Console.ReadLine();
+            }
+            return op;
+        }
+
+        private static bool IsValidOperator(string op)
+        {
+            string[] validOperators = { "a", "s", "m", "d", "p", "sin", "cos", "tan" };
+            foreach (string validOp in validOperators)
+            {
+                if (op == validOp)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static double ValidateNumberInput(Calculator calculator)
```

Comment:
Overall your code is well organized and readable. I'd just add more spacing between lines when the code blocks don't necessarily relate to each other. Here line 155 does something, then you could add an empty line, then 156 and 157 are related so it's not necessary. But I'm being picky. 😊